src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -3429,6 +3429,13 @@ - (void)_pruneEndpointsIn:(MTRDeviceDataValueDictionary)previousPartsListValue
         }
         [self _removeClusters:clusterPathsToRemove doRemoveFromDataStore:NO];
         [self.deviceController.controllerDataStore clearStoredClusterDataForNodeID:self.nodeID endpointID:endpoint];
+
+        if (_currentSubscriptionCallback) {
+            auto clusterStateCache = _currentSubscriptionCallback->GetClusterStateCache();
+            if (clusterStateCache) {
+                clusterStateCache->ClearAttributes(static_cast<EndpointId>(endpoint.unsignedLongLongValue));
```

Comment:
Is it safe to mutate clusterStateCache while not being on the Matter queue?

Reply:
No, good catch!  For some reason I thought this was on the Matter queue.

Reply:
Fixed.